OwnYourData/ClinicalTrials/ClinicalTrialsView.swift

```diff
@@ -6,47 +6,96 @@
-                                .bold()
-                            Text(trial.id ?? "No NCI ID")
-                                .bold()
-                                .foregroundStyle(.secondary)
-                            Divider()
-                            Text(trial.detailDescription ?? "-")
-                                .font(.caption)
-                        }
+            content
+                .onAppear {
+                    Task {
+                        // Start fetching trials when the view appears
+                        await fetchTrials()
                     }
                 }
-            }
-                .navigationTitle("NCI Trials")
-                .task {
-                    trials.append(contentsOf: (try? await loadTrials().data) ?? [])
-                }
                 .viewStateAlert(state: $viewState)
         }
     }
     
+    @ViewBuilder
+    private var content: some View {
+        if isLoading {
+            ProgressView()
+        } else if trials.isEmpty {
+            Text("No trials found.")
+        } else {
+            VStack(spacing: 10) {
+                // Horizontal bar for changing zip code and distance
+                searchBar
+                
+                List {
+                    ForEach(trials.indices, id: \.self) { index in
+                        TrialView(trial: trials[index], isCollapsedDescription: $collapseStates[index])
+                    }
+                }
+                .navigationTitle("NCI Trials")
+            }
+        }
+    }
     
-    private func loadTrials() async throws -> TrialResponse {
-        OpenAPIClientAPI.customHeaders = ["X-API-KEY": ""]
+    @ViewBuilder
+    private var searchBar: some View {
+        if !isLoading && !trials.isEmpty {
+            HStack {
+                TextField("Enter Zip Code", text: $zipCode)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                    .padding(.horizontal)
+                TextField("Enter Distance (mi)", text: $searchDistance)
+                    .textFieldStyle(RoundedBorderTextFieldStyle())
+                    .padding(.horizontal)
+                Button("Update Search") {
+                    Task {
+                        // Perform search with updated zip code and distance
+                        await fetchTrials()
+                    }
+                }
+                .padding(.horizontal)
+            }
+            .padding(.top)
+        }
+    }
+        
+    // Function to convert zip code to coordinates
+    private func getLocationFromZipCode(zipCode: String) async -> CLLocation? {
+        do {
+            let placemarks = try await geocoder.geocodeAddressString(zipCode)
+            if let location = placemarks.first?.location {
+                return location
+            }
+        } catch {
+            print("Error converting zip code to coordinates: \(error)")
+        }
+        return nil
+    }
```

Comment:
This function could still exist despite the usage of SpeziLocation. It might be more elegant to move it into an extension of `CLGeocoder`.